content/blog/2019/performance-best-practices/index.md

```diff
@@ -0,0 +1,110 @@
+
+Let's see how to use this test environment to analyze the data plane performance of Istio 1.2.
+
+For measuring sidecar proxy latency, we look at the 99th, 90th, and 50th percentiles for an increasing number of concurrent connections, keeping request throughput (RPS) constant.
+
+We found that with 16 concurrent connections and 1000 RPS, Istio adds **3ms** over the baseline (P50) when a request travels through both a client and server proxy. At 64 concurrent connections, Istio adds **12ms** over the baseline, but with Mixer disabled (`nomixer_both`), Istio only adds **7ms**.
+
+{{< image  width="75%" ratio="60%"
+    link="./latency_p50.png"
+    alt="Istio sidecar proxy, 50th percentile latency"
+    title="Istio sidecar proxy, 50th percentile latency"
+    caption=""
+    >}}
+
+In the 90th percentile, with 16 concurrent connections, Istio adds **6ms**; with 64 connections, Istio adds **20ms**.
+
+{{< image width="75%" ratio="60%"
+    link="./latency_p90.png"
+    alt="Istio sidecar proxy, 90th percentile latency"
+    title="Istio sidecar proxy, 90th percentile latency"
+    caption=""
+    >}}
+
+Finally, in the 99th percentile, with 16 connections, Istio adds **10ms** over the baseline. At 64 connections, Istio adds **25ms** with Mixer, or **10ms** without Mixer.
+
+{{< image  width="75%" ratio="60%"
+    link="./latency_p99.png"
+    alt="Istio sidecar proxy, 99th percentile latency"
+    title="Istio sidecar proxy, 99th percentile latency"
+    caption=""
+    >}}
+
+For CPU usage, we measured with an increasing request throughput (RPS), and a constant number of concurrent connections. We found that Envoy's maximum CPU usage at 3000 RPS, with Mixer enabled, was **1.2 vCPUs**. At 1000 RPS, one Envoy uses approximately half of a CPU.
+
+{{< image  width="75%" ratio="60%"
+    link="./cpu_max.png"
+    alt="Istio sidecar proxy, max CPU usage"
+    title="Istio sidecar proxy, max CPU usage"
+    caption=""
+    >}}
+
+## Summary
+
+In the process of benchmarking Istio's performance, we learned several key lessons:
+
+*   use an environment that mimics production
+*   focus on data plane traffic
+*   measure against a baseline
+*   increase concurrent connections as well as total throughput.
+
+For Istio 1.2, we found that on average, with 1000 RPS across 16 connections, Istio adds just **3 milliseconds** of latency over the baseline.
+
+A final note: Istio's performance depends on your specific setup and traffic load. Because of this variance, make sure your test setup accurately reflects your production workloads. To try out the benchmarking scripts, head over [to the Istio Tools repo](https://github.com/istio/tools/tree/76e3cb2488303316c8511a3ebe9676828c9d4765/perf/benchmark).
```

Comment:
I suggest you make this final note a "tip" call out.
```suggestion
{{< tip >}} 
Istio's performance depends on your specific setup and traffic load. Because of this variance, make sure your test setup accurately reflects your production workloads. To try out the benchmarking scripts, head over [to the Istio Tools repo](https://github.com/istio/tools/tree/76e3cb2488303316c8511a3ebe9676828c9d4765/perf/benchmark).
{{< /tip >}}
```